enigma-swing/src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java

```diff
@@ -207,7 +208,31 @@ private void updateList() {
 		this.classListModel = classListModel;
 		classList.setModel(classListModel);
 
-		currentSearch = su.asyncSearch(searchField.getText(), (idx, e) -> SwingUtilities.invokeLater(() -> classListModel.insertElementAt(e, idx)));
+		// handle these search result like minecraft scheduled tasks to prevent
+		// flooding swing buttons inputs etc with tons of (possibly outdated) invocations
+		record Order(int idx, SearchEntryImpl e) {}
```

Comment:
You can declare records in methods? Whoaaaa

Reply:
yeah, you can actually declare local classes way before records are a thing and have name like SearchDialog$1Order